packages/tests/packages/rest/read.spec.ts

```diff
@@ -0,0 +1,300 @@
+                ...fakePost(),
+                title: lorem.words(3).slice(0, 23),
+                user: user.id
+            })
+            return instance.ctx.orm.em.persistAndFlush(entity)
+        })
+    )
+
+    let response = await client
+        .get(`/api/resources/posts?perPage=10&page=1&search=new title 1`)
+        .send(userDetails)
+
+    expect(response.status).toBe(200)
+    expect(response.body.total).toBe(1)
+    expect(response.body.data[0].title).toBe(titles[0])
+
+    response = await client
+        .get(`/api/resources/posts?perPage=10&page=1&search=new title 2`)
+        .send(userDetails)
+
+    expect(response.status).toBe(200)
+    expect(response.body.total).toBe(1)
+    expect(response.body.data[0].title).toBe(titles[1])
+
+    response = await client
+        .get(`/api/resources/posts?perPage=10&page=1&search=new title`)
+        .send(userDetails)
+
+    expect(response.status).toBe(200)
+    expect(response.body.total).toBe(2)
+
+    response.body.data.map((d: { title: string }, i: number) => {
+        expect(d.title).toBe(titles[i])
+    })
+})
+test(`find all related resources (posts) `, async () => {
+    const instance = await setup()
+    const client = Supertest(instance.app)
+
+    const userDetails = {
+        email: Faker.internet.exampleEmail(),
+        full_name: Faker.name.findName(),
+        password: 'password'
+    }
+
+    const user = instance.ctx.orm.em.create('User', userDetails)
+    await instance.ctx.orm.em.persistAndFlush(user)
+
+    const [user1, user2] = await Promise.all(
+        [Faker.internet.exampleEmail(), Faker.internet.exampleEmail()].map(
+            email => {
+                const entity = instance.ctx.orm.em.create('User', {
+                    ...userDetails,
+                    email
+                })
+                return instance.ctx.orm.em.persistAndFlush(entity)
+            }
+        )
+    )
+
+    await Promise.all(
+        Array.from({ length: 3 }).map(() => {
+            const entity = instance.ctx.orm.em.create('Post', {
+                ...fakePost(),
+                title: lorem.words(3).slice(0, 23),
+                user: user.id
+            })
+            return instance.ctx.orm.em.persistAndFlush(entity)
+        })
+    )
+
+    await Promise.all(
+        Array.from({ length: 4 }).map(() => {
+            const entity = instance.ctx.orm.em.create('Post', {
+                ...fakePost(),
+                title: lorem.words(3).slice(0, 23),
+                user: user.id
+            })
+            return instance.ctx.orm.em.persistAndFlush(entity)
+        })
+    )
+
+    // this is to test for the first user
+
+    let response = await client
+        .get(`/api/resources/users/1/posts`)
+        .send({ ...userDetails, email: user1.toJSON().email })
+
+    expect(response.status).toBe(200)
+    expect(response.body.total).toBe(3)
+
+    response.body.data.map((d: { user_id: number }) => {
+        expect(d.user_id).toBe(1)
```

Comment:
This will not work for mongodb. Mongodb IDs are going to be strings. Make this dynamic